bin/check-temperature.rb

```diff
@@ -0,0 +1,88 @@
+#! /usr/bin/env ruby
+#
+#   check-temperature
+#
+# DESCRIPTION:
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: socket
+#   lm-sensors
+#
+# USAGE:
+#
+# NOTES:
+#
+# LICENSE:
+#   Jordan Beauchamp beauchjord@gmail.com
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'socket'
+
+#
+# Sensors
+#
+class CheckTemperature < Sensu::Plugin::Check::CLI
+  #
+  # Setup variables
+  #
+  def initialize
+    super
+    @crit_temp = []
+    @warn_temp = []
+  end
+
+  #
+  # Generate output
+  #
+
+  def usage_summary
+    (@crit_temp + @warn_temp).join(', ')
+  end
+
+  #
+  # Main function
+  #
+  def run
+    raw = `sensors`
+    sections = raw.split("\n\n")
+    metrics = {}
+    sections.each do |section|
+      section.split("\n").drop(1).each do |line|
+        begin
+          key, value = line.split(':')
+          key = key.downcase.gsub(/\s/, '')
+          if key[0..3] == 'temp' || key[0..3] == 'core'
```

Comment:
I suggest also including `loc` as a key here.

Example from one of my machines:
```plain
root@machine:~# sensors
i350bb-pci-0500
Adapter: PCI adapter
loc1:         +63.0°C  (high = +120.0°C, crit = +110.0°C)

coretemp-isa-0000
Adapter: ISA adapter
Physical id 0:  +32.0°C  (high = +86.0°C, crit = +100.0°C)
Core 0:         +28.0°C  (high = +86.0°C, crit = +100.0°C)
<REST OMITTED>
```